workbench-core/accounts/src/services/costCenterService.ts

```diff
@@ -34,6 +36,67 @@ export default class CostCenterService {
     this._dynamoDbService = dynamoDbService;
   }
 
+  /**
+   * Soft Delete Cost Center
+   * @param request - request for deleting cost center
+   * @returns void
+   */
+  public async softDeleteCostCenter(request: DeleteCostCenterRequest): Promise<void> {
+    const costCenterHaveProjects = await this._doesCostCenterHaveProjects(request.id);
+    if (costCenterHaveProjects) {
+      throw Boom.conflict(
+        `CostCenter ${request.id} cannot be deleted because it has project(s) associated with it`
+      );
+    }
+
+    const getCostCenterResponse = (await this._dynamoDbService
+      .get(buildDynamoDBPkSk(request.id, resourceTypeToKey.costCenter))
+      .execute()) as GetItemCommandOutput;
+
+    if (getCostCenterResponse.Item === undefined) {
+      throw Boom.notFound(`Could not find cost center ${request.id}`);
+    }
+
+    try {
+      await this._dynamoDbService
+        .update(buildDynamoDBPkSk(request.id, resourceTypeToKey.costCenter), {
+          item: { resourceType: `${this._resourceType}_deleted` }
+        })
+        .execute();
+    } catch (e) {
+      throw Boom.internal('Unable to delete CostCenter');
+    }
+  }
+
+  /**
+   * Update costCenter
+   * @param request - request for updating cost center
+   * @returns CostCenter object with updated attributes
+   */
+  public async updateCostCenter(request: UpdateCostCenterRequest): Promise<CostCenter> {
+    const currentDate = new Date().toISOString();
+
+    const updatedCostCenter = {
+      name: request.name,
+      description: request.description,
```

Comment:
If the user did not pass `description` in, would this evaluate to `undefined` and update the DDB item as `undefined` description?

Reply:
I just double check, and this works as expected. If a user does not pass in a `description` the field does not get changed in DDB.

Reply:
It's because of this setting
https://github.com/aws-solutions/solution-spark-on-aws/blob/22c4e9677dd0c7c89d47ba98fda748b27422e849/workbench-core/base/src/aws/helpers/dynamoDB/dynamoDBService.ts#L319